Add docs for MFA feature

Roma36 opened the pull request:
# Description of the change

This PR adds documentation for MFA feature.

## Checklist

Please make sure that the proposed change checks all the boxes below before requesting a review:

- [x] I have reviewed the [guidelines for contributing](https://github.com/spacelift-io/user-documentation/blob/main/CONTRIBUTING.md) to this repository.
- [x] The preview looks fine.
- [x] The tests pass.
- [x] The commit history is clean and meaningful.
- [x] The pull request is opened against the `main` branch.
- [x] The pull request is no longer marked as a draft.
- [x] You agree to license your contribution under the [MIT license](https://github.com/spacelift-io/user-documentation/blob/main/LICENSE) to Spacelift (not required for Spacelift employees).
- You have updated the navigation files correctly:
    - [ ] No new pages have been added, or;
    - [ ] Only _nav.yaml_ has been updated because the changes only apply to SaaS, or;
    - [ ] Only _nav.self-hosted.yaml_ has been updated because the changes only apply to Self-Hosted, or;
    - [x] Both _nav.yaml_ and _nav.self-hosted.yaml_ have been updated.

If the proposed change is ready to be merged, please request a review from `@spacelift-io/solutions-engineering`. Someone will review and merge the pull request.

_Spacelift employees should request reviews from the relevant engineers and are allowed to merge pull requests after they got at least one approval._

Thank you for your contribution! 🙇

jmfontaine commented on docs/product/security/mfa.md:
```diff
@@ -0,0 +1,42 @@
+### Adding Security Keys
+
+- Click on the _Enable_ button to activate MFA.
+![](./personal-settings-enable-mfa.png)
```
Screenshots must be stored in the `docs/assets/screenshots` folder.

jmfontaine commented on docs/product/security/mfa.md:
```diff
@@ -0,0 +1,42 @@
+
+Admins can view and delete security keys for any user within the organization to maintain the organization's security integrity.
+
+![](./organization-settings-mfa.png)
```
Screenshots must be stored in the `docs/assets/screenshots` folder.

jmfontaine reviewed (dismissed):
Except for the misplaced screenshots, it looks good to me.

Roma36 commented:
@jmfontaine I have addressed your comments and rebased this branch, but for some reason, one external link is failing check: 
https://git.alpinelinux.org/aports/tree/testing/spacectl/APKBUILD

jmfontaine reviewed (approved):
Looks good to me.
